src/main/java/Duke.java

```diff
@@ -1,10 +1,133 @@
 public class Duke {
+
+    /**
+     * The main method is the entry point of the program. It prompts the user for their name,
+     * instantiates a TaskMaster object, and handles user commands.
+     *
+     * @param args Command-line arguments passed to the program.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        showWelcomeMessage();
+        showHelpMessage();
+
+        IO inOut = new IO();
+
+        String userName = getUserName(inOut);
+
+        TaskMaster compileItems = new TaskMaster();
+
+        instantiateList(inOut, userName, compileItems);
+
+    }
+
+    /**
+     * The markAsDone method marks a task as done.
+     *
+     * @param compileItems The TaskMaster object that contains the task list.
+     * @param index        The index of the task to mark as done.
+     */
+    static void markAsDone(TaskMaster compileItems, int index) {
+        TaskManager toBeDone = compileItems.getTask(index);
+        boolean hasDukeMarked = toBeDone.markAsDone();
+        if (hasDukeMarked == true) {
+            System.out.println("You have completed this task. Good job!!");
+        } else {
+            System.out.println("This task has been marked before. Go on to another task.");
+        }
+        toBeDone.printTaskAndStatus(index);
+    }
+
+    /**
+     * The markAsNotDone method marks a task as not done.
+     *
+     * @param compileItems The TaskMaster object that contains the task list.
+     * @param index        The index of the task to mark as not done.
+     */
+    static void markAsNotDone(TaskMaster compileItems, int index) {
+        TaskManager toBeDone = compileItems.getTask(index);
+        boolean hasDukeUnMarked = toBeDone.markAsUndone();
+        if (hasDukeUnMarked == true) {
+            System.out.println("I would strongly encourage you to start on this.");
+        } else {
+            System.out.println("This task was never completed. I would strongly encourage you to start on it.");
+        }
+        toBeDone.printTaskAndStatus(index);
+    }
```

Comment:
Perhaps, you can merge these two methods into one, because there are several repetitive codes